modules/isofs/zso.c

```diff
@@ -16,13 +16,14 @@ u32 ziso_total_block;
 // block buffers
 u8 *ziso_tmp_buf = NULL;
 
-void ziso_init(ZISO_header *header, u32 first_block)
+void ziso_init(ZISO_header *header)
 {
     // read header information
     ziso_align = header->align;
     ziso_idx_start_block = -1;
-    // calculate number of blocks without using uncompressed_size (avoid 64bit division)
-    ziso_total_block = ((((first_block & 0x7FFFFFFF) << ziso_align) - sizeof(ZISO_header)) / 4) - 1;
+    // calculate number of blocks without using 64 bit division
+    u32 *total_bytes_p = (u32 *)&(header->total_bytes);
+    ziso_total_block = (total_bytes_p[0] >> 11) | ((total_bytes_p[1] & 0x7ff) << 21);
```

Comment:
Trust gcc 11. It's smart enough to do this for you :). I just did a small test:

```
u32 test_div(u64 bytes)
{
	return bytes / 2048;
}
```

compiles into:

```
test_div:
	sll	$2,$5,21
	srl	$4,$4,11
	jr	$31
	or	$2,$2,$4
```

But it's much more readable code (your code and comment don't even mention we are doing a divide by 2048).

Reply:
Weird, iirc the last time I tried to do that division it would ask me to link against some 64bit arithmetic library. But if it no longer does that then yes it's better.